### PR TITLE
Mtdsa 27674 - Implement TaxYear based Multiple Schema Definitions

### PR DESCRIPTION
**Pull request by DavouJ**

### CODE REVIEW

File: app/v1/residentialPropertyDisposals/retrieveAll/model/response/RetrieveAllResidentialPropertyCgtResponse.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import play.api.libs.json.OWrites
+import utils.JsonWritesUtil.writesFrom
+import v1.residentialPropertyDisposals.retrieveAll.def1.model.response.Def1_RetrieveAllResidentialPropertyCgtResponse
+
+trait RetrieveAllResidentialPropertyCgtResponse
+
+object RetrieveAllResidentialPropertyCgtResponse {
+
+  //implicit val format: OFormat[RetrieveAllResidentialPropertyCgtResponse] = Json.format[RetrieveAllResidentialPropertyCgtResponse]
```
Comment (benryan1790) [inline app/v1/residentialPropertyDisposals/retrieveAll/model/response/RetrieveAllResidentialPropertyCgtResponse.scala:27]: Is this needed? Remove please if not

File: test/v1/residentialPropertyDisposals/createAmendNonPpd/def1/Def1_CreateAmendCgtResidentialPropertyDisposalsValidatorSpec.scala [hunk 2]
```diff
@@ -326,16 +328,24 @@ class CreateAmendCgtResidentialPropertyDisposalsValidatorFactorySpec
   private def validator(nino: String, taxYear: String, body: JsValue) =
     validatorFactory.validator(nino, taxYear, body)
 
-  private val MINIMUM_YEAR = 2020
-  MockedAppConfig.minimumPermittedTaxYear returns MINIMUM_YEAR
+//  private val MINIMUM_YEAR = 2020
+//  MockedAppConfig.minimumPermittedTaxYear returns MINIMUM_YEAR
```
Comment (benryan1790) [inline test/v1/residentialPropertyDisposals/createAmendNonPpd/def1/Def1_CreateAmendCgtResidentialPropertyDisposalsValidatorSpec.scala:332]: remove please

**Review by benryan1790 — CHANGES_REQUESTED** [review]

File: app/v1/residentialPropertyDisposals/retrieveAll/model/response/RetrieveAllResidentialPropertyCgtResponse.scala (hunk 1 above)
Comment (benryan1790) [inline app/v1/residentialPropertyDisposals/retrieveAll/model/response/RetrieveAllResidentialPropertyCgtResponse.scala:27]: is this needed? Please remove if not

Comment (benryan1790) [inline app/v1/residentialPropertyDisposals/retrieveAll/model/response/RetrieveAllResidentialPropertyCgtResponse.scala:27]: is this needed? remove if not please

File: app/v1/residentialPropertyDisposals/deleteCgtPpdOverrides/DeleteCgtPpdOverridesService.scala [hunk 3]
```diff
@@ -14,44 +14,42 @@
...
+      "SERVER_ERROR" -> InternalError,
+      "SERVICE_UNAVAILABLE" -> InternalError
     )
 
-    val extraTysErrors: Map[String, MtdError] = Map(
+    val extraTysErrors = Map(
       "INVALID_CORRELATION_ID" -> InternalError,
-      "TAX_YEAR_NOT_SUPPORTED" -> RuleTaxYearNotSupportedError,
-      "NOT_FOUND"              -> NotFoundError
+      "NOT_FOUND" -> NotFoundError,
```
[the remark this answers is not in the document]
Comment (DavouJ) [reply app/v1/residentialPropertyDisposals/deleteCgtPpdOverrides/DeleteCgtPpdOverridesService.scala:48]: No, I think i was slightly confused when making these changes and hadn't read the documentation properly. Reverting all the errors in all services back to what they were originally.

**Review by benryan1790 — APPROVED** [review]

**Pull request merged**